app/controllers/dashboard_controller.rb

```diff
@@ -10,21 +10,44 @@ def sheets
   end
 
   def save
-    deletes = Array.wrap(params['delete']).concat(params['add'].keys)
+    return head :bad_request if invalid_day_periods?
```

Comment:
`:unprocessable_entity` sounds more appropriate to me considering we're dealing with a kind of param validation. Another thing is about being clear on which error we're dealing with (returning a clear message).

Reply:
Yeah, very good tip! Thanks!